src/EA.Weee.RequestHandlers/Admin/GetProducerDetails/RemoveSmallProducerHandler.cs

```diff
@@ -0,0 +1,46 @@
+﻿namespace EA.Weee.RequestHandlers.Admin.GetProducerDetails
+{
+    using Core.Admin;
+    using Prsd.Core.Mediator;
+    using Requests.Admin;
+    using Security;
+    using System.Threading.Tasks;
+    using Weee.Security;
+
+    public class RemoveSmallProducerHandler : IRequestHandler<RemoveSmallProducer, RemoveSmallProducerResult>
+    {
+        private readonly IWeeeAuthorization authorization;
+        private readonly IRemoveProducerDataAccess removeProducerDataAccess;
+
+        public RemoveSmallProducerHandler(IWeeeAuthorization authorization, IRemoveProducerDataAccess removeProducerDataAccess)
+        {
+            this.authorization = authorization;
+            this.removeProducerDataAccess = removeProducerDataAccess;
+        }
+
+        public async Task<RemoveSmallProducerResult> HandleAsync(RemoveSmallProducer request)
+        {
+            authorization.EnsureCanAccessInternalArea();
+            authorization.EnsureUserInRole(Roles.InternalAdmin);
+
+            var producerSubmissions =
+                await removeProducerDataAccess.GetProducerSubmissionsForRegisteredProducer(request.RegisteredProducerId);
+
```

Comment:
@stevelee-civica  you wont need any of the producerSubmissions code.

Reply:
updated